### PR TITLE
Highlight multiline step arguments

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+### Fixed
+
+* Add highlighting support for multiline step arguments
```
Comment (valfirst) [inline CHANGELOG.md:10]:
```suggestion
* Highlight multiline step arguments
```

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/util/Splitter.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+public final class Splitter
+{
+    private Splitter()
+    {
+    }
+
+    public static List<String> split(String text)
+    {
+        return Stream.of(text.split("\\R", -1)).collect(Collectors.toList());
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/util/Splitter.java:34]: missing unit tests

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/service/SemanticTokensService.java [hunk 3]
```diff
@@ -40,40 +45,101 @@ public SemanticTokensService(StepDefinitionResolver stepDefinitionResolver)
...
+        }
+
+        private void addToken(Integer from, Integer length)
+        {
+            semanticTokens.addAll(List.of(
+                from,
+                length,
+                0,
+                0
+            ));
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/service/SemanticTokensService.java:137]: `java.util.Collections#addAll`

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,12 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+### Fixed
+
+* Highlight multiline step arguments
```
Comment (ikalinin1) [inline CHANGELOG.md:10]:
```suggestion
* Highlighting of multiline step arguments.
```

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/model/Step.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+/*-
```
Comment (ikalinin1) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/model/Step.java:1]: is this minus expected?

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/service/SemanticTokensService.java [hunk 6]
```diff
@@ -40,40 +45,99 @@ public SemanticTokensService(StepDefinitionResolver stepDefinitionResolver)
...
+                    }
 
-                semanticTokens.addAll(List.of(
-                    from - previousFrom,
-                    length,
-                    0,
-                    0
-                ));
+                    int limit = argumentLengths.size() - 1;
+                    int shift = argumentLengths.subList(0, limit).stream()
```
Comment (ikalinin1) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/service/SemanticTokensService.java:102]:
```suggestion
                    int shift = argumentLengths.stream().limit()
```

**Review by ikalinin1 — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (uarlouski) [reply CHANGELOG.md:10]: https://github.com/vividus-framework/vividus-studio/pull/345#discussion_r1101276032 :)

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/model/Step.java (hunk 5 above)
Comment (uarlouski) [reply vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/model/Step.java:1]: yes, this is how maven plugin works

**Pull request merged**